### PR TITLE
TEST: Azure DELETE

### PR DESCRIPTION
**Pull request by nicolas2bert**
This PR adds tests for Azure Delete

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAzure.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                    s3.deleteObject({
+                        Bucket: azureContainerName,
+                        Key: keyName,
+                    }, err => {
+                        assert.equal(err, null, 'Expected success ' +
+                            `but got error ${err}`);
+                        s3.getObject({ Bucket: azureContainerName,
+                        Key: keyName }, err => {
+                            assert.strictEqual(err.code, 'NoSuchKey',
+                            'Expected error but got success');
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAzure.js]: also check to make sure not on azure when get directly.

**Comment by nicolas2bert** [comment]
For now we return error: `400 - InvalidArgument` if deleting specifying a version.

File: tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAzure.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                });
+            });
+
+            afterEach(done => {
+                azureClient.deleteBlob(azureContainerName, azureObject, err => {
+                    done(err);
+                });
+            });
+
+            it('should return error 501 NotImplemented on deleting an ' +
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAzure.js]: can you add a test for the aws backend as well?

**Comment by LaurenSpiegel** [comment]
As discussed, please add a test trying to delete a specific version and test that the result is no such key and then we can merge.

**Comment by nicolas2bert** [comment]
@ironman-machine try

**Pull request merged**